src/main/java/org/mockito/internal/junit/StrictStubsTestListener.java

```diff
@@ -0,0 +1,71 @@
+package org.mockito.internal.junit;
+
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.exceptions.base.MockitoAssertionError;
+import org.mockito.internal.creation.settings.CreationSettings;
+import org.mockito.internal.invocation.finder.AllInvocationsFinder;
+import org.mockito.internal.util.collections.ListUtil;
+import org.mockito.invocation.Invocation;
+import org.mockito.invocation.MatchableInvocation;
+import org.mockito.listeners.StubbingLookUpListener;
+import org.mockito.mock.MockCreationSettings;
+import org.mockito.stubbing.Stubbing;
+
+import java.util.*;
+
+import static org.mockito.Mockito.mockingDetails;
+import static org.mockito.internal.util.collections.ListUtil.filter;
+
+/**
+ * Test listener implementation that fails when there are unused stubbings
+ */
+class StrictStubsTestListener implements MockitoTestListener {
+
+    private final Map<Object, MockCreationSettings> mocks = new HashMap<Object, MockCreationSettings>();
+
+    public void beforeTest(Object testClassInstance, String testMethodName) {
+        //TODO init mocks and validate mockito usage is duplicated in the listeners, refactor or make sure all is tested
+        MockitoAnnotations.initMocks(testClassInstance);
+    }
+
+    public void afterTest(Throwable testFailure) {
+        if (testFailure == null) {
```

Comment:
@szczepiq what about returning asap? This avoids indentation and improves readability.
```java
if (testFailure!=null){
   return;
}
```